File: 15-key-vault/python-sdk/key_vault.py
"""
Cost Katana Key Vault: Provider Keys (Python)

Run: python 19-key-vault/python-sdk/key_vault.py
"""

import os
import requests

API_BASE = "https://cost-katana-backend.store/api"
API_KEY = os.getenv("COST_KATANA_API_KEY")

def create_provider_key(provider, api_key, name):
    response = requests.post(
        f"{API_BASE}/key-vault/provider-keys",
        json={"provider": provider, "apiKey": api_key, "name": name},
        headers={"Authorization": f"Bearer {API_KEY}"}
    )
    response.raise_for_status()
    print(f"✅ Created {provider} key")
    return response.json()["data"]

def list_provider_keys():
    response = requests.get(
        f"{API_BASE}/key-vault/provider-keys",
        headers={"Authorization": f"Bearer {API_KEY}"}
    )
    response.raise_for_status()
    print(f"📋 Found {len(response.json()['data'])} keys")
    return response.json()["data"]

if __name__ == "__main__":
    print("🥷 Key Vault: Provider Keys\n")
    create_provider_key("openai", "sk-proj-...", "Production Key")
    list_provider_keys()

File: 15-key-vault/npm-package/key-vault.ts
/**
 * Cost Katana Key Vault: Provider Keys Management
 * 
 * Run: npx ts-node 19-key-vault/npm-package/key-vault.ts
 */

import axios from 'axios';

const API_BASE = 'https://cost-katana-backend.store/api'\;
const API_KEY = process.env.COST_KATANA_API_KEY;

async function createProviderKey(provider: string, apiKey: string, name: string) {
  const response = await axios.post(
    `${API_BASE}/key-vault/provider-keys`,
    { provider, apiKey, name },
    { headers: { 'Authorization': `Bearer ${API_KEY}` } }
  );
  console.log(`✅ Created ${provider} key: ${response.data.data.id}`);
  return response.data.data;
}

async function listProviderKeys() {
  const response = await axios.get(
    `${API_BASE}/key-vault/provider-keys`,
    { headers: { 'Authorization': `Bearer ${API_KEY}` } }
  );
  console.log(`📋 Found ${response.data.data.length} provider keys`);
  return response.data.data;
}

async function main() {
  console.log('🥷 Key Vault: Provider Keys\n');
  await createProviderKey('openai', 'sk-proj-...', 'Production Key');
  await listProviderKeys();
}

if (require.main === module) main();
export { createProviderKey, listProviderKeys };
